crates/red_knot_module_resolver/src/typeshed.rs

```diff
@@ -21,9 +37,71 @@ mod tests {
         assert!(functools_module_stub.is_file());
 
         let mut functools_module_stub_source = String::new();
-        functools_module_stub.read_to_string(&mut functools_module_stub_source)?;
+        functools_module_stub
+            .read_to_string(&mut functools_module_stub_source)
+            .unwrap();
 
         assert!(functools_module_stub_source.contains("def update_wrapper("));
-        Ok(())
+    }
+
+    #[test]
+    fn typeshed_vfs_consistent_with_vendored_stubs() {
+        let vendored_typeshed_stubs = VendoredFileSystem::new(TYPESHED_ZIP_BYTES).unwrap();
+
+        let mut empty_iterator = true;
+        for entry in walkdir::WalkDir::new(&*VENDORED_TYPESHED_DIR).min_depth(1) {
+            empty_iterator = false;
+            let entry = entry.unwrap();
+            let absolute_path = entry.path();
+            let file_type = entry.file_type();
+
+            let relative_path = absolute_path.strip_prefix(&*VENDORED_TYPESHED_DIR).unwrap();
+            let vendored_path = VendoredPath::new(relative_path.to_str().unwrap());
+
+            assert!(vendored_typeshed_stubs.exists(vendored_path));
+
+            let vendored_path_kind = vendored_typeshed_stubs
+                .metadata(vendored_path)
+                .unwrap()
+                .kind();
+            assert_eq!(vendored_path_kind.is_directory(), file_type.is_dir());
+        }
+        assert!(!empty_iterator);
+    }
+
+    #[test]
+    fn typeshed_versions_consistent_with_vendored_stubs() {
```

Comment:
Nit: I would consider moving the tests closer to the implementation they're testing.